apps/duck-web/src/net/ws.ts

```diff
@@ -0,0 +1 @@
+export const openWs = (url: string, bearer: string) => new WebSocket(url, ['duck.v1', `bearer.${bearer}`]);\n",
```

Comment:
One-click DI wrapper + empty-bearer guard:

```suggestion
export type WsFactory = (url: string, protocols?: string | string[]) => WebSocket;
export const openWs = (mkWs: WsFactory = (u,p) => new WebSocket(u,p)) =>
  (url: string, bearer?: string) => {
    const protocols = ['duck.v1', ...(bearer ? [`bearer.${bearer}`] : [])];
    return mkWs(url, protocols);
  };
```